src/NuGet.Core/NuGet.Commands/RestoreCommand/LockFileBuilder.cs

```diff
@@ -360,6 +358,53 @@ private static void AddProjectFileDependenciesForPackageReference(PackageSpec pr
             }
         }
 
+        private void AddProjectCentralTransitiveDependencyGroupsForPackageReference(PackageSpec project, LockFile lockFile, IEnumerable<RestoreTargetGraph> targetGraphs)
+        {
+            // TargetFrameworkInformations should have only distinct FrameworkNames
+            var centralPackageVersionsPerFramework = project.TargetFrameworks.ToDictionary(tfmi => tfmi.FrameworkName, tfmi => tfmi.CentralPackageVersions);
+
+            foreach (var targetGraph in targetGraphs.OrderBy(graph => graph.Framework.ToString(), StringComparer.Ordinal))
```

Comment:
RestoreTargetGraph is per TFM/RID combo. We should only be considering the ridless target graphs, as those are the only dependencies that get written to the nuspec, as those are the user declared ones. 

So exclude `targetGraph.RuntimeIdentifier` where it's null or empty (not sure if you need to check null or empty, see what the rest of the code is doing).

Reply:
Updated

Reply:
Updated to exclude the not null RID graphs